### PR TITLE
core: helper: Only abort if server returns bad request

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

File: core/services/helper/main.py [hunk 1]
```diff
@@ -317,7 +317,7 @@ def detect_service(port: int) -> ServiceInfo:
             "127.0.0.1", port=port, path="/", timeout=1.0, method="GET", follow_redirects=10
         )
         log_msg = f"Detecting service at port {port}"
-        if response.status != http.client.OK:
+        if response.status == http.client.BAD_REQUEST:
```
Comment (joaoantoniocardoso) [inline core/services/helper/main.py:320]: could you explain the reason behind this?

Comment (joaoantoniocardoso) [reply core/services/helper/main.py:320]: shouldn't the servers give a 200 when valid?

Comment (patrickelectric) [reply core/services/helper/main.py:320]: In the commit body:
> Sometimes it can return a redirect or any other, register service will take care for us

Comment (patrickelectric) [reply core/services/helper/main.py:320]: My case was moving openvscode to vscode where the / was not an OK but something

Comment (joaoantoniocardoso) [reply core/services/helper/main.py:320]: okay, it looks prone to false positives, but it might be better than false negatives.

**Pull request merged**